File: project/app/src/main/java/com/marcosvbras/empresas/views/listeners/BaseViewModelCallback.java
package com.marcosvbras.empresas.views.listeners;

import android.os.Bundle;

public interface BaseViewModelCallback {

    void showErrorDialog(String message);
    void showErrorDialog(int message);
    void openActivity(Class<?> activity, boolean finishCurrentActivity);
    void openActivity(Class<?> activity, Bundle bundle, boolean finishCurrentActivity);
    void setToolbar(int viewId, String title, boolean displayHomeAsUpEnabled);
    void setToolbarTitle(String title);

}

File: project/app/src/main/java/com/marcosvbras/empresas/views/listeners/BaseViewModelCallback.kt
package com.marcosvbras.empresas.views.listeners

import android.os.Bundle

interface BaseViewModelCallback {
    fun showErrorDialog(message: String)
    fun showErrorDialog(message: Int)
    fun openActivity(activity: Class<*>, finishCurrentActivity: Boolean)
    fun openActivity(activity: Class<*>, bundle: Bundle, finishCurrentActivity: Boolean)
    fun setToolbar(viewId: Int, title: String, displayHomeAsUpEnabled: Boolean)
    fun setToolbarTitle(title: String)
}